05-html-css/flexbox-vs-css-grid.md

```diff
@@ -0,0 +1,153 @@
+
+By contrast, Grid example above will always maintain 3 columns.  There are ways to adjust the number of columns for smaller screens, by using something like: `grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));` which will generate as many columns as needed with a minimum size of 200 pixels and a maximum of 1 fraction of the row.  
+
+So Flexbox always responds to smaller screen sizes, while Grid can be configured in a responsive manner.   
+
+# TODO Replace Codepen
+**Exercise**: Try to update the [Codepen](https://codepen.io/CheezItMan/pen/pLjKEp?editors=1100) of the Grid layout to make it responsive for smaller screens.  
+
+# TODO Replace Codepen
+You can find a solution [here](https://codepen.io/CheezItMan/pen/ZxbRom).
+
+## Grid and Flexbox Work Well **Together**
+
+Consider the following site:
+
+![Sample website using Grid and flexbox](imgs/css-grid-layout.jpg)
+
+How would you lay out this content?  One common approach is to lay out the larger page with a grid, and then lay out smaller sections with Flexbox.
+
+The grid could look like this:
+
+![sample Grid layout (img from: https://drive.google.com/file/d/1dHjtl_XcMCJN1KaDAvsI5EwdLqUIzcpU/view?usp=sharing )](imgs/grid-layout.png)
+
+# TODO Replace Codepens
+**Exercise:** Take a look at the above grid and try to recreate the basic grid design with a header at the top aside on the left and main on the right with a footer crossing the bottom in [this codepen](https://codepen.io/CheezItMan/pen/vRNjbq?editors=1100).  Only the container elements are listed with and background colors are added to make the elements easier to see.
+
+You can find a solution [here](https://codepen.io/CheezItMan/pen/yKYjjM?editors=1100).
+
+Then Flexbox could lay out individual elements like:
+
+![Header](imgs/header-flexbox.png)
+
+Of course you could also lay out the subregions of the site with Grid as well, but if content is only along one axis, Flexbox usually requires less work.  
+
+**Question** How could you lay out the header with Flexbox so the button is on the right?
+
+## So do I use Flexbox or Grid?
+
+Ask yourself a couple of questions:
+- Do I only need to lay content out by row **or** by column?
+  - Use Flexbox
+- Do I need to control layout with both row **and** column?
+  - Use Grid
+- Is it important that items "line up" with each other?
+  - Use Grid
+
+## Browser Support
+
+It should be said that CSS Grid is newer than Flexbox and has less support.  However as of March 12, 2018 [88% of web traffic supports Grid](https://caniuse.com/#feat=css-grid).  This means if you need to support older browsers, you will need to consider alternatives to Grid.  There are strategies to create fallback styles to handle older browsers which we will not cover here, but you can [read about them](https://rachelandrew.co.uk/css/cheatsheets/grid-fallbacks).  At Ada we will use the modern Chrome browser as our standard and Chrome fully supports Grid.  
```

Comment:
I think I would not even include this section. We've talked about browser support a couple other places, I think covering it more is going to scare them off.

Reply:
You'd have to remove that bit from the conclusion as well.